lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+        teaming_policy.policy = vim.StringPolicy(inherited=False, value=self.teaming_load_balancing)
+        teaming_policy.failureCriteria = vim.dvs.VmwareDistributedVirtualSwitch.FailureCriteria()
+        results['failure_detection'] = self.teaming_failure_detection
+        if self.teaming_failure_detection == "link_status_only":
+            teaming_policy.failureCriteria.checkBeacon = vim.BoolPolicy(inherited=False, value=False)
+        elif self.teaming_failure_detection == "beacon_probing":
+            teaming_policy.failureCriteria.checkBeacon = vim.BoolPolicy(inherited=False, value=True)
+        # Deprecated since VI API 5.1. The system default (true) will be used
+        teaming_policy.reversePolicy = vim.BoolPolicy(inherited=False, value=True)
+        results['notify'] = self.teaming_notify_switches
+        teaming_policy.notifySwitches = vim.BoolPolicy(inherited=False, value=self.teaming_notify_switches)
+        # this option is called 'failback' in the vSphere Client
+        # rollingOrder also uses the opposite value displayed in the client
+        results['failback'] = self.teaming_failback
+        teaming_policy.rollingOrder = vim.BoolPolicy(inherited=False, value=not self.teaming_failback)
+        if self.teaming_failover_order_active or self.teaming_failover_order_standby:
+            teaming_policy.uplinkPortOrder = vim.dvs.VmwareDistributedVirtualSwitch.UplinkPortOrderPolicy()
+            teaming_policy.uplinkPortOrder.activeUplinkPort = self.teaming_failover_order_active
+            teaming_policy.uplinkPortOrder.standbyUplinkPort = self.teaming_failover_order_standby
+            results['failover_active'] = self.teaming_failover_order_active
+            results['failover_standby'] = self.teaming_failover_order_standby
+        pg_spec.defaultPortConfig.uplinkTeamingPolicy = teaming_policy
 
-    def state_create_dvspg(self):
+        # Monitoring - NetFlow
+        results['netflow_enabled'] = self.netflow_enabled
+        result = self.check_netflow_policy_config(pg_config=None)
+        pg_spec.defaultPortConfig.ipfixEnabled = result[0]
+
+        # TODO: Traffic filtering and marking
+
+        # Misc. - Block all ports
+        results['block_all_ports'] = self.block_all_ports
+        result = self.check_blocked_policy_config(pg_config=None)
+        pg_spec.defaultPortConfig.blocked = result[0]
+
+        if self.module.check_mode:
+            results['result'] = "Portgroup would be created"
+        else:
+            task = self.dv_switch.AddDVPortgroup_Task([pg_spec])
+            try:
+                wait_for_task(task)
+            except TaskError as invalid_argument:
+                self.module.fail_json(
+                    msg="Failed to create portgroup : %s" % to_native(invalid_argument)
+                )
+            results['result'] = "Portgroup created"
+
+        self.module.exit_json(**results)
+
+    @staticmethod
+    def get_api_port_binding_mode(mode):
```

Comment:
I'm trying to understand the logic here: If mode == earlyBinding, return 'static', but if it is static, return 'earlyBinding'? Why exactly? Shouldn't you always return what the API is expecting?

Reply:
Ah, so the goal is to do a back/forth translation between the values you want the user to use/see, and the actual API values. Sure

Could probably be improved by just using dicts instead of all the if/else clauses, but that's just details

Reply:
If you have a recommendation, I can implement. I use such functions in other modules as well.